datastore/src/main/java/com/ds/datastore/Utilities.java

```diff
@@ -23,14 +28,21 @@
 
     @Retry(name = "retry")
     public HttpResponse<String> createConnection(String address, JsonObject jso, String serverAddress, Long id, String requestType) throws Exception{
+        String requestID;
+        ServletRequestAttributes attr = (ServletRequestAttributes) RequestContextHolder.getRequestAttributes();
+        if(attr != null){
+             HttpServletRequest currentReq = attr.getRequest();
+             requestID = currentReq.getAttribute("requestID").toString();
+        }else{
+            requestID = String.valueOf(UUID.randomUUID());
```

Comment:
This is perfectly fine - just want to point out its not technically necessary since the listener will be setting a uuid if none is present.